### DISCUSSION TITLE
Handle multiple changie instances withing the same repo?

### DISCUSSION THREAD

**Discussion by marcosnils** [miniscruff/changie#506] [Ideas] [answered]
In Dagger, we have multiple changie instances within the same repo. For example here https://github.com/dagger/dagger/tree/main/sdk/go and here https://github.com/dagger/dagger/tree/main. This puts an extra overhead to the user to make sure they're in the correct folder when running `changie new`. It'd be awesome if somehow this cloud be consolidated and add some sort of `includes` or `references` to multiple changie instances within a single place to help with this situation. 

Thoughts?

cc @gerhard

**Comment by miniscruff** [comment, marked as answer]
Have you seen the proposal here? https://github.com/miniscruff/changie/discussions/502#discussioncomment-6456761 I believe it is related.

**Comment by Blacksmoke16** [comment]
I'd like to pitch my use case as well for this. The gist of it is I have a many-repo setup in which there is a monorepo, but unlike a normal monorepo, changes that are merged are then synced out to dedicated mirrors (repositories) for each project within the monorepo. Additionally each project is tagged/released from the mirrors and _not_ the monorepo. As such i'd like for the changie instances to live within each project itself rather than the monorepo root.

I read thru #502, but don't think it'll exactly work how I need. Mainly given it seems to assume the `.changes` directory lives at the monorepo root and not within each project itself.

**Reply by miniscruff** [reply]
Would you be able to export a tree or something about your setup? First idea is the "include" option on batch https://changie.dev/cli/changie_batch/ to separate your unreleased directories. Might be a bit awkward though.

**Reply by Blacksmoke16** [reply]
Sure yea, the structure is like:

```txt
src/
└── components
    ├── clock
    ├── console
    ├── contracts
    ├── dependency_injection
    ├── dotenv
    ├── event_dispatcher
    ├── framework
    ├── image_size
    ├── mercure
    ├── mime
    ├── negotiation
    ├── routing
    ├── serializer
    ├── spec
    └── validator
```
Where each directory has a `CHANGELOG.md` and its own `src/`, etc. Everything within each component directory is synced to a dedicated repository. There is no top level `CHANGELOG.md` file either. I'll prob have more useful feedback after I actually play with things a bit, but current thought is something like `make -C` so then could do `changie -C src/components/clock new` which would at least allow you to stay within the monorepo root and not have to manually cd to/from different directories if working on multiple components at once. But the `.changes` and changelog files could still be component specific.

**Reply by miniscruff** [reply]
yea, you could also create a `.changes/unreleased` per sub directory and include each with `--project`. So like `changie batch auto --include components/clock/.changes/unrelease --project clock`? But I can't say I have tried this before.

**Reply by Blacksmoke16** [reply]
I thought about this more and came to the conclusion that I'm okay with `.changes` being only in the monorepo as it's the source of truth about everything. This means I can just leverage the existing `projects` feature and that seems to be working quite well so far.

One question I did have that wasn't obviously clear; is there a way to have each project directory within `.changes` to have its own `unreleased` directory instead of them all being in one with a prefixed change file?

**Reply by miniscruff** [reply]
Unfortunately there is not an option to customize that I believe I had it specced out in the v2 config format but not in the current setup.
